src/pages/lessons/lesson-player-machine.ts

```diff
@@ -0,0 +1,109 @@
+import {Machine} from 'xstate'
+
+interface PlayerStateSchema {
```

Comment:
felt like this state machine was related to the lesson page more than the player specifically so I moved it here. We might consider consolidating all of this and the overlays into the player later?